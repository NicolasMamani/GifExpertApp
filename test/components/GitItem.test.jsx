import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('pruebas en <GifItem />', () => {
    const title = 'Messi';
    const url = 'https://localhost/messi.jpg';
    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL indicado y el ALT indicado', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        // screen.debug();
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
        // console.log(screen.getByRole('img').src);
    });
});
