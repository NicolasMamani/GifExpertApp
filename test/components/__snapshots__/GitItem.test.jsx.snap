// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Messi"
      src="https://localhost/messi.jpg"
    />
    <p>
      Messi
    </p>
  </div>
</div>
`;
